File: solutions/java/resistor-color-duo/1/src/main/java/ResistorColorDuo.java
public class ResistorColorDuo {

    private static final String[] COLORS = {
        "black",
        "brown",
        "red",
        "orange",
        "yellow",
        "green",
        "blue",
        "violet",
        "grey",
        "white"
    };

    public int value(String[] colors) {
        int first = colorCode(colors[0]);
        int second = colorCode(colors[1]);
        return first * 10 + second;
    }

    private int colorCode(String color) {
        for (int i = 0; i < COLORS.length; i++) {
            if (COLORS[i].equals(color)) {
                return i;
            }
        }
        return -1;
    }
}

File: solutions/python/resistor-color-duo/1/resistor_color_duo.py
# resistor_color_duo.py

# Mapping of color to digit
COLOR_CODES = {
    "black": 0,
    "brown": 1,
    "red": 2,
    "orange": 3,
    "yellow": 4,
    "green": 5,
    "blue": 6,
    "violet": 7,
    "grey": 8,
    "white": 9
}

def value(colors):
    """
    Given a list of color names, return the two-digit resistor value
    corresponding to the first two colors.
    """
    first_two = colors[:2]
    digits = [str(COLOR_CODES[color.lower()]) for color in first_two]
    return int("".join(digits))
